### PR TITLE
flho-142 Company-name step

### PR DESCRIPTION
**Pull request by bendiggle**
Added company-name step:	
* Added step to routes config,
* Added fields to field config with toggle fields,
* Added translations for fields, pages and validation
* Added acceptance tests for the step,
* Added page helper for the tests

### CODE REVIEW

File: apps/new-dealer/fields/index.js [hunk 1]
```diff
@@ -1,6 +1,43 @@
 'use strict';
 
 module.exports = {
+  'company': {
```
Comment (joefitter) [inline apps/new-dealer/fields/index.js]: `'`s aren't needed

File: apps/new-dealer/fields/index.js [hunk 2]
```diff
@@ -1,6 +1,43 @@
...
+    mixin: 'radio-group',
+    validate: ['required'],
+    className: ['form-group'],
+    options: [{
+      value: 'true',
+      label: 'fields.company.options.true',
```
Comment (joefitter) [inline apps/new-dealer/fields/index.js]: ooh ooh - watch this space - PR in template mixins for this to be looked up magically!

**Comment by bendiggle** [comment]
@joefitter This has been updated!
@JoeChapman @gxxm

File: apps/new-dealer/acceptance/pages/company-name.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  url: 'company-name',
+  'company-group': '#company-group',
+  company: '#company-true',
+  'company-name': '#company-name',
+  'company-house-number': '#company-house-number',
+  'sole-trader': '#company-false',
+  'sole-trader-name': '#sole-trader-name',
+
+  content: {
+    name: 'Home Office',
```
Comment (joefitter) [inline apps/new-dealer/acceptance/pages/company-name.js]: can we make this more archer specific please?

File: apps/new-dealer/views/partials/company.html [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<div id="company-name-panel" class="reveal" aria-hidden="false">
+  <div class="panel-indent">
+    {{#input-text}}company-name{{/input-text}}
+    {{#input-text}}company-house-number{{/input-text}}
```
Comment (joefitter) [inline apps/new-dealer/views/partials/company.html]: and here

File: apps/new-dealer/views/partials/company.html [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<div id="company-name-panel" class="reveal" aria-hidden="false">
+  <div class="panel-indent">
+    {{#input-text}}company-name{{/input-text}}
```
Comment (joefitter) [inline apps/new-dealer/views/partials/company.html]: otherwise there are 2 places this field is tied to type - field config and the template

**Comment by bendiggle** [comment]
@joefitter Updated

**Comment by bendiggle** [comment]
@JoeChapman @gxxm

**Comment by bendiggle** [comment]
@joefitter Actually Don't merge friend. The renderField doesn't work. Needs to be `input-text`

**Comment by joefitter** [comment]
it should work :'(

**Comment by bendiggle** [comment]
:'( 
screen shot 2016-09-20 at 16 45 21

**Comment by joefitter** [comment]
MEH

**Comment by joefitter** [comment]
https://github.com/UKHomeOffice/hof-template-partials/blob/master/views/step.html#L4 - it needs to be called in scope - the template has access to an array of field objects. I will fix the mixin, its NOT VERY GOOD

File: apps/new-dealer/translations/src/en/validation.json [hunk 6]
```diff
@@ -1,4 +1,16 @@
 {
+  "company": {
+    "required": "Please select whether you're applying on behalf of a company or a sole trader"
```
Comment (joefitter) [inline apps/new-dealer/translations/src/en/validation.json]: I thought we didn't say please?

Comment (bendiggle) [inline apps/new-dealer/translations/src/en/validation.json]: @joefitter I didn't know that! These validation messages are by me and not done by a designer

**Comment by joefitter** [comment]
LGTM - we will need to revisit this at some point as dependent does funny (not funny) things.

**Comment by joefitter** [comment]
do we say please? I thought we didn't say please?

**Comment by bendiggle** [comment]
@joefitter I've removed all the Pleases!

**Comment by bendiggle** [comment]
@JoeChapman @gxxm

**Comment by joefitter** [comment]
very pleasing

**Comment by joefitter** [comment]
I am pleased.

File: apps/new-dealer/translations/src/en/validation.json (hunk 6 above)
Comment (joefitter) [inline apps/new-dealer/translations/src/en/validation.json]: Well, mr Diggle, you are far more polite than any designer I have ever met

**Comment by bendiggle** [comment]
This pleases me

**Pull request merged**